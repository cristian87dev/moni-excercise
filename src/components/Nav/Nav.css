.nav-menu {
  display: flex;
  flex-grow: 1;

  ul {
    flex-grow: 1;
    display: flex;
    justify-content: space-around;
    align-items: center;
  }
}

.nav-item {
  padding: 0.5rem 0;
  list-style: none;

  @media (min-width: 768px) {
    padding: 0;
  }
}

.link-item {
  color: var(--font-color);
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
}
