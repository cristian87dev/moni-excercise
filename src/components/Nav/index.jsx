import React from 'react';
import { Link } from 'react-router-dom';

import './Nav.css';

const Nav = () => (
  <nav className="nav-menu">
    <ul>
      <li className="nav-item">
        <Link className="link-item" to="/pedi-tu-prestamo">Ped√≠ tu Prestamo</Link>
      </li>
      <li className="nav-item">
        <Link className="link-item" to="/user-admin">User Admin</Link>
      </li>
    </ul>
  </nav>
);

export default Nav;
