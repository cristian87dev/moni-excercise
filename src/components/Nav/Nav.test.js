import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom'; // To prevent Link dependency error.
import renderer from 'react-test-renderer';
import Nav from './index';

it('Nav renders without crashing', () => {
  const component = renderer.create(
    <Router>
      <Nav />
    </Router>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
