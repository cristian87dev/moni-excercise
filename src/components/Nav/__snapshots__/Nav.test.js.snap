// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders without crashing 1`] = `
<nav
  className="nav-menu"
>
  <ul>
    <li
      className="nav-item"
    >
      <a
        className="link-item"
        href="/pedi-tu-prestamo"
        onClick={[Function]}
      >
        Ped√≠ tu Prestamo
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="link-item"
        href="/user-admin"
        onClick={[Function]}
      >
        User Admin
      </a>
    </li>
  </ul>
</nav>
`;
